### PR TITLE
Move Snapshot APIs in-tree

### PR DESCRIPTION
**Pull request by xing-yang**
This PR moves snapshot APIs in-tree.
The design proposal is here: https://github.com/kubernetes/community/pull/1662

**What this PR does / why we need it**:
This PR moves snapshot APIs in-tree.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #543

**Special notes for your reviewer**:

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -364,6 +370,7 @@ var defaultKubernetesFeatureGates = map[utilfeature.Feature]utilfeature.FeatureS
 	DynamicProvisioningScheduling:               {Default: false, PreRelease: utilfeature.Alpha},
 	VolumeSubpathEnvExpansion:                   {Default: false, PreRelease: utilfeature.Alpha},
 	KubeletPluginsWatcher:                       {Default: false, PreRelease: utilfeature.Alpha},
+	VolumeSnapshot:                              {Default: true, PreRelease: utilfeature.Alpha},
```
Comment (dixudx) [inline pkg/features/kube_features.go]: Normally we don't enable alpha feature by default until it is beta.

File: pkg/printers/internalversion/describe.go [hunk 2]
```diff
@@ -1332,6 +1334,127 @@ func describePersistentVolumeClaim(pvc *api.PersistentVolumeClaim, events *api.E
...
+
+func (d *VolumeSnapshotDescriber) Describe(namespace, name string, describerSettings printers.DescriberSettings) (string, error) {
+	c := d.Storage().VolumeSnapshots(namespace)
+
+	vs, err := c.Get(name, metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+
+	events, _ := d.Core().Events(namespace).Search(legacyscheme.Scheme, vs)
```
Comment (dixudx) [inline pkg/printers/internalversion/describe.go]: Better not ignore the error here.

**Review by dixudx — COMMENTED** [review]
Thanks for your contribution.

Please add some tests for the describers and printers.

/cc @kubernetes/sig-storage-pr-reviews

File: pkg/features/kube_features.go (hunk 1 above)
Comment (xing-yang) [reply pkg/features/kube_features.go]: Got it.  Will fix it.

**Comment by xing-yang** [comment]
Close this one because the Snapshot API changes are already included in the Snapshot Controller PR https://github.com/kubernetes/kubernetes/pull/64663.

**Pull request closed without merging**